src/dbnode/storage/bootstrap.go

```diff
@@ -143,9 +158,24 @@ func (m *bootstrapManager) Bootstrap() error {
 		}
 		m.Unlock()
 
-		if !currPending {
-			break
+		if currPending {
+			// NB(r): Requires another bootstrap.
+			continue
+		}
+
+		if bootstrapErr != nil {
+			// NB(r): Last bootstrap failed, since this could be due to transient
+			// failure we retry the bootstrap again. This is to avoid operators
+			// needing to manually intervene for cases where failures are transient.
+			m.log.Warn("retrying bootstrap after backoff",
+				zap.Duration("backoff", bootstrapRetryInterval),
+				zap.Int("numRetries", i+1))
+			m.sleepFn(bootstrapRetryInterval)
+			continue
```

Comment:
This should probably have a limit on number of retries, otherwise can get stuck here if the last bootstrapper is constantly failing